### ISSUE TITLE
Background/foreground techniques update or remove C23, 25

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/wcag#490]
Jared tweeted me about background/foreground techniques C23, C25.

@davidmacd What are your thoughts on https://www.w3.org/TR/WCAG20-TECHS/C23.html … and https://www.w3.org/TR/WCAG20-TECHS/C25.html … These seem significantly out-of-date and irrelevant to modern browsers (it references FF3 and IE6!) and CSS/OS color techniques. If so, they should be removed or updated, right?

**Comment by mraccess77** [comment]
I disagree they are irrelevant.  They are very relevant to both Firefox 62 and to IE 11.  I’d encourage people to reach out to the low vision task force about things like this as people with low vision are the best ones to answer to say what is still relevant to us.  

In particular pages can specify, foreground, background, none, or both.   Settings in the browsers allow users to customize colors only or customize colors and overwrite what is specified by the webpage.  There is also a failure technique for only specifying background but not color or vice versa.  That failure technique is also still relevant.    So one browser option is to allow users can set their own colors for content unless the author has specified colors in which case the authors colors are used.  This would allow a banner and the main part of a page to look different if no colors were specified in the main portion of the page.

I’d also mention that color replacement is also available in Adobe Acrobat and Adobe Reader and used with similar challenges there.

The high contrast extension in Chrome is not equivalent and does not provide the level of specification of color changes needed by people with low vision.  It is merely a page level filter and often can cause text to have less contrast.

**Comment by jared-w-smith** [comment]
Jonathan -

Thank you for the additional insight into this. With few browsers still providing the functionality to specify colors that don't overwrite author-specified colors - and with this only one of numerous color-override options available (and arguably a rather inferior one compared to other methods) - I do wonder how useful and common these are for users with low vision. I'd be happy to hear feedback from the LVTF on this.

I think it's relatively rare for authors to not specify page foreground/background colors, but also define foreground/background colors for all of the other notable page regions, as C23 requires. It's also rather difficult to test this technique, except in these specific browsers. 

The normative text states "Foreground and background colors can be selected by the user." Authors specifying a page foreground and background color do not preclude this ability (as is evidenced by https://www.w3.org/TR/WCAG20-TECHS/G156.html). While C23 and C25 are both written as sufficient techniques, they read more like failure techniques (i.e., "If you define page colors then you don't pass."). It seems that this could cause confusion.

All of this combined makes these techniques rather odd and potentially confusing. But if they're still highly useful for users, they should be maintained. If they are, the user agent notes (https://www.w3.org/WAI/WCAG20/Techniques/ua-notes/css#C23) are woefully out-of-date and should probably be updated.

I appreciate consideration of this. Being AAA techniques, I'm not overly concerned about this, but I think it worth reviewing.

**Comment by Ryladog** [comment]
Might be worth updating surely, but not removing.
